File: C++/杂项/i128.cpp
__int128 read() {
    string s;
    cin >> s;
    __int128 x = 0;
    int f = 1, i = 0;
    if (s[0] == '-') f = -1, i = 1;
    for (; i < s.size(); i++) x = x * 10 + (s[i] - '0');
    return x * f;
}

void print(__int128 x) {
    if (x < 0) {
        cout << '-';
        x = -x;
    }
    if (x > 9) print(x / 10);
    cout << (int)(x % 10);
}

File: Python/杂项/i128.py
# 重载输入运算符以支持__int128类型
def read_int(s: str) -> int:
    return int(s)


# 重载输出运算符以支持__int128类型
def write_int(x: int) -> str:
    return str(x)


# Python int is arbitrary precision, so it covers C++ __int128 use cases.
